### ISSUE TITLE
Removing "typo" from ranking rules does not disable typo tolerance while searching

### ISSUE THREAD

**Issue by bhavyalatha26** [meilisearch/MeiliSearch#1103]
**Describe the bug**
To disable the typo tolerance while searching, it is recommended to remove the rule **typo** from the list of ranking rules. However this does not work. Even after removing it from the ranking rules, the results still seem to apply typo tolerance.

**To Reproduce**
1. Using Client update the settings
```python
client = meilisearch.Client(url, apiKey=api_key)
client.get_index(index_name).update_ranking_rules([
        "words",
        "proximity",
        "attribute",
        "wordsPosition",
        "exactness"
    ])
```
2. Restart the server
3. See that the results still consider typo tolerance

**Expected behavior**
The search should ignore typo tolerance and follow only the required rules.

**Screenshots**
![error-2](https://user-images.githubusercontent.com/13586777/100107026-3b98c200-2e8f-11eb-9b89-1b3360d91ee7.png)

**Additional context**
I have also tried to use only **['exactness']** for raking rules, but still the typo tolerance is being applied.

**Comment by theo-lubert** [comment]
I agree with @bhavyalatha26, this should be possible:
- by removing the `typo` ranking rule **(preferred)**
- through an index setting
- through a search option
- It could also make sense to configure it for specific fields (enabled on title/description, but disabled on id/ean/trackingNumber for example). But that feels like another feature to me

Algolia, for exemple, allows this with `typoTolerance: false` (https://www.algolia.com/doc/api-reference/api-parameters/typoTolerance/)

**Comment by qdequele** [comment]
Hello, This issue will be fixed by a new feature/setting. 

I will close this issue in favor of our [public roadmap](https://roadmap.meilisearch.com/c/61-policy-for-prefix-search?utm_medium=social&utm_source=portal_share). 

I invite everyone interested in this feature to upvote it on the roadmap.

**Comment by sirdharma** [comment]
@qdequele Thank you for your answer and for sharing your public roadmap, that's awesome 👍 

Nonetheless, I am confused by your linked idea: how would the proposed policy for prefix search work exactly?

In a real-world use case, let's assume we have an index that contains only a UUID-type attribute. Is it possible to configure the index so that typo-tolerance is not applied at all?

Let's then consider we have an index that contains a UUID-type attribute and a name attribute. Is it possible to configure the index' attributes so that typo-tolerance is disabled on the first attribute, but enabled on the second?

As already mentioned by @theo-lubert in https://github.com/meilisearch/MeiliSearch/issues/1103#issuecomment-734343137, Algolia provides both [`typoTolerance`](https://www.algolia.com/doc/api-reference/api-parameters/typoTolerance/) and [`disableTypoToleranceOnAttributes`](https://www.algolia.com/doc/api-reference/api-parameters/disableTypoToleranceOnAttributes/) for that exact purpose.

My apologies if I failed to see how prefix search policy would solve that. If that's the case, I'll definitely up-vote it 😄

**Comment by theo-lubert** [comment]
I agree with @sirdharma , it would make a lot more sense to me if this was field specific, like Algolia's "disableTypoToleranceOnAttributes", and also not limited to the prefix.
Btw, maybe this could also improve indexing performance.

@qdequele maybe this should be another feature on the roadmap.

**Comment by Sembiance** [comment]
Was this ever added into meilisearch?

The three things preventing me from adopting meilisearch are:
- sortBy
- Disable typo tolerance per field, entirely or at search time
- Delete by query

**Comment by curquiza** [comment]
Hello @Sembiance! Thanks for your feedback, it might interest @gmourier 😇 

Also, about the SortBy, it will be implemented in v0.22.0 available in the middle of September, here is the tracking issue: https://github.com/meilisearch/MeiliSearch/issues/1581

**Comment by curquiza** [comment]
About disable the typo tolerance: there is a workaround by using the phrase search (available in v0.21.0, the official release is next week). You have to surround each query word by `"` when sending the query to MeiliSearch.
Ex: if the user types `meili search` you will have to send `"meili" "search"` to the search engine

However there are some limits to this workaround: using the `"` the search engine will no try to gather words or to split them.
Ex: if the user types `meili search`, you send `"meili" "search"`. But the search engine will not retrieve documents containing `meilisearch`, only documents containing `meili` and `search`.

FYI, I open a feature request about it: https://github.com/meilisearch/product/discussions/176

**Comment by Sembiance** [comment]
@curquiza  AWESOME workaround! That'll work great for the time being. Thanks!
